### DISCUSSION TITLE
The best way to choose selectors for the multilanguage application

### DISCUSSION THREAD

**Discussion by mikalaikiveisha** [webdriverio/webdriverio#10085] [Q&A]
According to https://webdriver.io/docs/selectors/ is preferable to use selectors those contain the element and the appropriate text like `$('button=Submit')`. What is the best way for choosing selectors if my application supports different languages (e.g. English, Spanish, French etc.)? Rely on testid attributes or something else?

**Comment by christian-bromann** [comment]
Thanks for asking the question @mikalaikiveisha !

How does your application manage multi language support? I wonder if you can re-use that someway in your tests too.

> Rely on testid attributes or something else?

Totally, that works too! WebdriverIO recommends using testids if you don't mind adding these attributes for testing.

**Reply by mikalaikiveisha** [reply]
The language is selected according to the region set in system settings also there is a possibility to change the language through the application menu. Thanks!
